aws/lambda/node/cache-invalidator/lambda.js

```diff
@@ -0,0 +1,69 @@
+const zlib = require('zlib');
+const AWS = require('aws-sdk'); // eslint-disable-line import/no-unresolved
+
+const cloudfront = new AWS.CloudFront();
+
+exports.handler = async event => {
+  // Log event
+  // console.log(JSON.stringify(event))
+  const payload = Buffer.from(event.awslogs.data, 'base64');
+  const parsed = JSON.parse(zlib.gunzipSync(payload).toString('utf8'));
+  console.log('Decoded payload:', JSON.stringify(parsed));
+
+  const logEvent = parsed.logEvents[0];
+  const {message} = logEvent;
+  const messageParts = message.split('\t');
+  const messageObject = messageParts[2];
+  const parsedMessageObject = JSON.parse(messageObject);
+  console.log(parsedMessageObject);
+  const {distributionId} = parsedMessageObject;
+  const callerReference = 'cache-invalidator-lambda-' + new Date().toISOString();
+  const {host} = parsedMessageObject;
+  const isBeta = host.split('.')[1] === 'beta';
+
+  const getDistributionId = new Promise((resolve, reject) => {
+    if (isBeta) {
```

Comment:
I think you could also write this without another promise by using await on the inner promise.

Something like this

```javascript
if (isBeta) {
   try {
     const data = await cloudfront.listDistributions({MaxItems: '1000'}).promise()
     // set getDistributionId from data
   } catch err {
   }
} else {
  getDistributionId = distributionId;
}
```

not 100% sure about this though, as I'm new to this syntax